From 3e880faa4f76eb7aa61893f5de73fb0bb9b6a4ab Mon Sep 17 00:00:00 2001
From: Waldemar Brodkorb <wbx@openadk.org>
Date: Wed, 11 Oct 2017 13:01:42 +0200
Subject: h8300: remove dead code

As h8/300 architecture does not use deprecated syscalls
after reintroduction into Linux upstream, this part of
the code in vfork.S is always unused.
---
 libc/sysdeps/linux/h8300/vfork.S | 23 -----------------------
 1 file changed, 23 deletions(-)

diff --git a/libc/sysdeps/linux/h8300/vfork.S b/libc/sysdeps/linux/h8300/vfork.S
index 85b733e..1e041c6 100644
--- a/libc/sysdeps/linux/h8300/vfork.S
+++ b/libc/sysdeps/linux/h8300/vfork.S
@@ -15,28 +15,6 @@
 	.hidden	__vfork
 	.type	__vfork,@function
 __vfork:
-#ifdef __NR_vfork
-	mov.l	@sp+, er1
-	sub.l	er0,er0
-	mov.b	#__NR_vfork,r0l
-	trapa	#0
-
-	mov.l	#-4096, er2
-	cmp.l 	er0,er2
-	bcs     fix_errno
-	jmp	@er1		/* don't return, just jmp directly */
-fix_errno:
-	neg.l	er0
-# if !defined(__PIC__)
-	mov.l	er0,@_errno
-# else
-	mov.l	@(_errno@GOTOFF,er5),er2
-	mov.l	er0,@er2
-# endif
-	sub.l	er0,er0
-	dec.l	#1,er0
-	jmp	@er1		/* don't return, just jmp directly */
-#else
 	mov.l	@sp+,er2		/* er2 = return address */
 	mov.l	#vfork_args,er1
 	sub.l	er0,er0
@@ -63,7 +41,6 @@ vfork_args:
 	.long	0
 	.long	0
 	.long	0
-#endif
 
 weak_alias(__vfork,vfork)
 libc_hidden_def(vfork)
-- 
cgit v0.11.2-3-g2eb4
